receiver/gitmetricsreceiver/README.md

```diff
@@ -0,0 +1,34 @@
+# Git Metrics Receiver
+
+<!-- status autogenerated section -->
+| Status        |           |
+| ------------- |-----------|
+| Stability     | [development]: metrics   |
+| Distributions | [] |
+
+[development]: https://github.com/open-telemetry/opentelemetry-collector#development
+<!-- end autogenerated section --> 
+
+The Git Metrics receiver generates metrics from Git about Git repositories and 
+organizations from various vendor endpoints. It is intended to be used
+when the collector is deployed as an agent, scraping over a given interval of
+time.
+
+## Getting Started
+
+The collection interval is defaulted to 30 seconds across the board. 
+
+```yaml
+gitmetrics:
```

Comment:
I'm still not sure if we got the receiver's name right - `gitmetrics`. I have a couple thoughts:

1. I'd rather get rid of the `metrics` part in the receiver name. It feels redundant to me.
2. I wonder if we should emphasize that this receiver's scope is "repository" metrics, and not CI/CD metrics that many Git vendors offer. Or perhaps we shouldn't specify the scope of the receiver in its name, as we may want to change it in the future? :thinking:

Brainstorming some names:
- `git` - someone might think it's metrics from the `git` binary? :thinking: 
- `gitprovider` - explicitly saying it's not metrics from `git` binary and only about Git itself, but from Git providers, whatever they provide
- `gitrepo` - narrows down the receiver's scope to only "Git repository" metrics
- `githealth` - narrows down the receiver's scope to only "Git repository health" metrics

Reply:
This is a good callout. 

> Or perhaps we shouldn't specify the scope of the receiver in its name, as we may want to change it in the future?

I lean towards this being the case. This current set is just initial, and thinking towards the future, industry definitions like DORA, and the OTEP, I do think that it'd be better to look at this as a `gitprovider` receiver instead of `gitmetrics`.

`gitprovider` has my vote 👍🏼